numpy/core/setup.py

```diff
@@ -605,6 +606,12 @@ def generate_api(ext, build_dir):
     config.add_include_dirs(join('src', 'umath'))
     config.add_include_dirs(join('src', 'npysort'))
 
+    # Need to add location of npy_math_internal.h, generated from src/npymath/npy_math_internal.h.src
+    npymath_build_dir = join('build', 'src.%s-%s.%s' % (sysconfig.get_platform(), sys.version_info[0], sys.version_info[1]),
```

Comment:
ah now I remember how this worked. You have to add a function to the sources line of the `add_extension`, that function receives the build directory as argument, see `generate_numpyconfig_h`

I guess one could add the include directory there as with the private header directory, add a new function just for the multiarray math include directory or move the header to the private header location.
Probably the first option  is simplest albeit a but ugly as the `generate_numpyconfig_h` technically has a different job ...